hazelcast/src/main/java/com/hazelcast/spi/impl/executionservice/InternalExecutionService.java

```diff
@@ -18,14 +18,27 @@
 
 import com.hazelcast.spi.ExecutionService;
 
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
 /**
  * The internal {@link ExecutionService}. Currently this method doesn't contains any additional methods, but
  * in the future they can be added here.
  */
 public interface InternalExecutionService extends ExecutionService {
 
+    ExecutorService getDurable(String name, boolean scheduled);
 
     void executeDurable(String name, Runnable command);
 
+    ScheduledFuture<?> scheduleDurable(String name, Runnable command, long delay, TimeUnit unit);
+
+    <V> ScheduledFuture<V> scheduleDurable(String name, Callable<V> command, long delay, TimeUnit unit);
+
+    ScheduledFuture<?> scheduleDurableWithRepetition(String name, Runnable command,
```

Comment:
This is the contract of `scheduleWithRepetition()`: 
`If any execution of this task takes longer than its period, then subsequent execution will be skipped.`

This `skip-on-overlap` strategy is the very reason why it has such unusual name: To indicate it has a different contract from `scheduleAtFixedRate` or `scheduleWithFixedDelay`.